platform/mbed_error_report.c

```diff
@@ -0,0 +1,271 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2006-2013 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <stdlib.h>
+#include <stdarg.h>
+#include "hal/serial_api.h"
+#include "hal/itm_api.h"
+#include "platform/mbed_error.h"
+#include "platform/mbed_error_report.h"
+
+#ifdef DEVICE_SERIAL
+extern int stdio_uart_inited;
+extern serial_t stdio_uart;
+#endif
+
+/* Converts a uint32 to hex char string */
+static void value_to_hex_str(uint32_t value, char *hex_str)
+{
+    char hex_char_map[] = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F'};
+    int i = 0;
+    
+    //Return without converting if hex_str is not provided
+    if(hex_str == NULL) return;
+        
+    for(i=7; i>=0; i--) {
+        hex_str[i] = hex_char_map[(value & (0xf << (i * 4))) >> (i * 4)];
+    }
+}
+
+/* Converts a uint32 to dec char string */
+static void value_to_dec_str(uint32_t value, char *dec_str)
+{
+    char dec_char_map[] = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9'};
+    int i = 0;
+    
+    //Return without converting if hex_str is not provided
+    if(dec_str == NULL) return;
+        
+    while(value > 0) {
+        dec_str[i++] = dec_char_map[value % 10];
+        value = value / 10;
+    }
+}
+
+//Helper function to get the current SP
+static unsigned int get_current_sp()
```

Comment:
I believe this one needs to take into account cortex-a (see the build failure).

Reply:
Yes, thanks for pointing that out, I'm looking into that. It appears there are no equivalent Mbed-os intrinsics for Cortex-A and so I'm looking into this.